File: example/src/main/java/ru/virgil/example/truck/TruckService.java
package ru.virgil.example.truck;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Service;
import ru.virgil.example.order.BuyingOrder;
import ru.virgil.example.system.SimpleJpaAccess;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class TruckService implements SimpleJpaAccess<Truck, UUID> {

    private final TruckRepository repository;

    public Truck get(UUID uuid) {
        return repository.findById(uuid).orElseThrow();
    }

    public List<Truck> getAll(BuyingOrder buyingOrder, int page, int size) {
        return repository.findAllByBuyingOrder(buyingOrder, PageRequest.of(page, size));
    }

    public Truck assignTruck() {
        return repository.findAll().stream().findAny().orElseThrow();
    }

    @Override
    public JpaRepository<Truck, UUID> getRepository() {
        return repository;
    }
}

File: example/src/main/java/ru/virgil/example/truck/TruckService.kt
package ru.virgil.example.truck

import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service
import ru.virgil.example.order.BuyingOrder
import java.util.*

@Service
class TruckService(private val repository: TruckRepository) {

    fun get(uuid: UUID): Truck = repository.findById(uuid).orElseThrow()

    fun getAll(buyingOrder: BuyingOrder, page: Int, size: Int): List<Truck> =
        repository.findAllByBuyingOrderContaining(buyingOrder, PageRequest.of(page, size))

    fun assignTruck(): Truck = repository.findAll().random()
}
